File: linearsearch.cpp
#include <iostream>
using namespace std;
int linearsearch(int arr[],int size,int target){
    for(int i=0;i<size;i++){
        if (arr[i]==target){
        return i;
    }}
    return -1;
}
int main(){
    int size;
    int target;
    cout<<"enter a target - ";
    cin>>target;
    cout<<"enter a sixe of the array - ";
    cin>>size;
    int number[size];
    cout <<"enter "<<size<<" array ";
    for( int i=0;i<size;i++)
    {
        cin>>number[i];
    }
    int result = linearsearch(number,size,target);
    if(result!=-1){
        cout<<"element found at index"<<result;
    }
    else{
        cout<<"element not found";}
}

File: LinearSearch.java
public class LinearSearch {

    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i; // return index when element is found
            }
        }
        return -1; // element not found
    }

    public static void main(String[] args) {
        int[] arr = {10, 25, 30, 45, 50};
        int target = 45;

        int index = linearSearch(arr, target);

        if (index != -1) {
            System.out.println("Element found at index: " + index);
        } else {
            System.out.println("Element not found");
        }
    }
}
